stan/math/prim/mat/fun/add_diag.hpp

```diff
@@ -2,57 +2,39 @@
 #define STAN_MATH_PRIM_MAT_FUN_ADD_DIAG_HPP
 
 #include <stan/math/prim/mat/fun/Eigen.hpp>
-#include <stan/math/prim/scal/err/check_finite.hpp>
-#include <stan/math/prim/scal/err/check_greater_or_equal.hpp>
-#include <stan/math/prim/scal/err/check_not_nan.hpp>
-#include <stan/math/prim/scal/fun/fmin.hpp>
+#include <stan/math/prim/scal/err/check_consistent_sizes.hpp>
+#include <stan/math/prim/scal/meta/scalar_seq_view.hpp>
 #include <stan/math/prim/scal/meta/return_type.hpp>
 #include <algorithm>
 
 namespace stan {
 namespace math {
 /**
- * Returns a Matrix with a constant added along the main diagonal.
+ * Returns a Matrix with values added along the main diagonal
  *
  * @tparam T_m type of element in Eigen::Matrix
- * @tparam T_a type of element to add along the diagonal
- *
- * @param mat a matrix of dimensions greater than 1, 1
- * @param to_add a constant to add along the diagonal
- * @return a matrix with a constant added along main diagonal
+ * @tparam T_a Type of element to add along the diagonal
+ * @param mat a matrix
+ * @param to_add (Sequence of) value(s) to add along the diagonal
+ * @return a matrix with to_add added along main diagonal
```

Comment:
Please doc with when it throws. Just for context, it would take me a few mins to reason through whenever the inputs are valid.